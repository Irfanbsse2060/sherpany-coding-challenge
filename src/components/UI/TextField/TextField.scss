@import "../../../styles/variables";


.text-field {
  text-align: left;
  width: 300px;

  &__label {
    display: block;
    font-size: $font-size-small;
    color: $scorpion;
    margin-bottom: 2px;
  }

  &__input {
    font-size: $font-size-base;
    border-radius: 4px;
    border: 2px solid $mercury;
    width: 100%;
    background: transparent;
    line-height: 16px;
    padding: 7px;

     &:focus{
      border-color: $color-brand;
      outline: 0;
    }
  }
}
