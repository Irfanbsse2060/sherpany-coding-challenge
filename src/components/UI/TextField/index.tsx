import React, {ChangeEvent} from 'react'
import './TextField.scss'

interface TextFieldProps {
    label: string
    name: string
    type?: string
    value: string | number
    handleChange: (event: ChangeEvent<HTMLInputElement>) => void
    testId?: string
    required?: boolean
}

export default function TextField(props: TextFieldProps) {
    const {type = 'text', required = false, testId, label, name, value, handleChange} = props
    return (
        <div className='text-field'>
            <label className='text-field__label' >{label}</label>
            <input
                className='text-field__input'
                data-testid={testId || `text_field_${name}`}
                name={name}
                type={type}
                value={value}
                required={required}
                onChange={handleChange}/>
        </div>
    )

}
