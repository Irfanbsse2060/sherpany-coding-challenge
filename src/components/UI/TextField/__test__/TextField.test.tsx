import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TextField from '../index'

describe('Text Field', ()=>{

    test('should be able to render', () => {
        const { container } = render(<TextField value='test' label='label' name='name' handleChange={jest.fn()}/>);
        expect(container).toMatchSnapshot()
    })

    test('should be able to render with required', () => {
        const { container } = render(<TextField value='test' label='label' name='name' required={true} handleChange={jest.fn()}/>);
        expect(container).toMatchSnapshot()
    });

    test('should call handle change method', () => {
        const mockHandleChange =  jest.fn()
        const { getByTestId } = render(<TextField value='test' label='label' name='name' required={true} handleChange={mockHandleChange}/>);
        fireEvent.change(getByTestId('text_field_name'), { target: { value: 'newValue' } })
        expect(mockHandleChange).toHaveBeenCalledTimes(1)
    });

})
