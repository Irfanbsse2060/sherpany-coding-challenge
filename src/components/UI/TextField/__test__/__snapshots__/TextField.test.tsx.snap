// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Field should be able to render 1`] = `
<div>
  <div
    class="text-field"
  >
    <label
      class="text-field__label"
    >
      label
    </label>
    <input
      class="text-field__input"
      data-testid="text_field_name"
      name="name"
      type="text"
      value="test"
    />
  </div>
</div>
`;

exports[`Text Field should be able to render with required 1`] = `
<div>
  <div
    class="text-field"
  >
    <label
      class="text-field__label"
    >
      label
    </label>
    <input
      class="text-field__input"
      data-testid="text_field_name"
      name="name"
      required=""
      type="text"
      value="test"
    />
  </div>
</div>
`;
